ngraph/core/src/preprocess/pre_post_process.cpp

```diff
@@ -52,62 +75,117 @@ static void propagate_layout(const std::shared_ptr<Node>& src, const std::shared
     }
 }
 
+using InternalPreprocessOp =
+    std::function<std::shared_ptr<ov::Node>(const std::vector<std::shared_ptr<ov::Node>>& nodes,
+                                            PreprocessingContext& context)>;
+
 /// \brief PreProcessStepsImpl - internal data structure
 struct PreProcessSteps::PreProcessStepsImpl {
     void add_scale_impl(const std::vector<float>& values) {
         m_actions.emplace_back(std::make_tuple(
-            [values](const std::shared_ptr<Node>& node) {
+            [values](const std::vector<std::shared_ptr<Node>>& nodes, PreprocessingContext&) {
+                OPENVINO_ASSERT(!nodes.empty(), "Internal error: Can't apply scale preprocessing for empty input.");
+                OPENVINO_ASSERT(nodes.size() == 1,
+                                "Can't apply scale preprocessing for multi-plane input. Suggesting to convert current "
+                                "image to RGB/BGR color format using 'convert_color'");
                 Shape shape;
                 if (values.size() == 1) {
                     shape = Shape{1};
                 } else {
-                    shape = construct_mean_scale_shape(node, values.size());
+                    shape = construct_mean_scale_shape(nodes[0], values.size());
                 }
                 auto constant = op::v0::Constant::create(element::f32, shape, values);
-                constant->set_friendly_name(node->get_friendly_name() + "/scale/Divide_Factor");
+                constant->set_friendly_name(nodes[0]->get_friendly_name() + "/scale/Divide_Factor");
 
-                auto new_op = std::make_shared<op::v1::Divide>(node, constant);
-                new_op->set_friendly_name(node->get_friendly_name() + "/scale/Divide");
-                propagate_layout(node, new_op);
+                auto new_op = std::make_shared<op::v1::Divide>(nodes[0], constant);
+                new_op->set_friendly_name(nodes[0]->get_friendly_name() + "/scale/Divide");
+                propagate_layout(nodes[0], new_op);
                 return new_op;
             },
             false));
     }
 
     void add_mean_impl(const std::vector<float>& values) {
         m_actions.emplace_back(std::make_tuple(
-            [values](const std::shared_ptr<Node>& node) {
+            [values](const std::vector<std::shared_ptr<Node>>& nodes, PreprocessingContext&) {
+                OPENVINO_ASSERT(!nodes.empty(), "Internal error: Can't apply mean preprocessing for empty input.");
+                OPENVINO_ASSERT(nodes.size() == 1,
```

Comment:
check for NV12 / I420 color format instead

Reply:
This is preprocessing of one 'original' input, which can be potentially represented as multiple nodes (like multiplane color format). I guess it is pretty safe to not allow most of preprocessing (except convert_element_type) until appropriate 'convert_color' is added